build.gradle.kts

```diff
@@ -544,17 +497,10 @@ afterEvaluate {
 	fun getChangelog(): String {
 		return file("CHANGELOG.md").readText()
 	}
-
-	fun getTaskForPublish(): TaskProvider<out DefaultTask> {
-		return if (releaseChannel.compress)
-			compressJar
-		else
-			tasks.shadowJar
-	}
 	
 	fun getFileForPublish(): RegularFile {
-		return if (releaseChannel.compress)
-			compressJar.get().getOutputJar()
+		return if (releaseChannel.deflation != JarShrinkingType.NONE)
```

Comment:
This can also just use the output from `compressJar`, since the task will always be `compressJar` now. The task it got the file from was supposed to match for both the functions.